feat(FR-1420): add deployment select component and pagination hook

agatha197 opened the pull request:
Resolves #4209 ([FR-1420](https://lablup.atlassian.net/browse/FR-1420))

# Add DeploymentSelect component with cursor-based pagination

This PR adds a new `DeploymentSelect` component for selecting deployments in the Chat interface. It includes:

- A new `DeploymentSelect.tsx` component with search functionality and pagination
- A new `useRelayCursorPaginatedQuery` hook for cursor-based pagination with Relay
- Added `endpointUrl` filter to the GraphQL schema for deployments
- Added translations for "Select Deployment" across all supported languages

The component provides a searchable dropdown with lazy-loading capabilities, displaying deployment names and allowing navigation to deployment details.

**Checklist:**

- [x] Documentation
- [ ] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

[FR-1420]: https://lablup.atlassian.net/browse/FR-1420?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

agatha197 committed:
feat(FR-1416): add deployment select component and pagination hook

ironAiken2 reviewed (changes requested):
It seems to need a restack.

agatha197 committed:
fix(FR-1420): fill out deployment namespace translations across all languages

All non-English language files now include the complete deployment namespace
(120 flat keys + 93 launcher keys). English text is used as fallback where
translations don't exist yet. Previously translated keys (e.g., SelectDeployment)
are preserved.

This resolves the review feedback that deployment strings were missing in
other languages.

agatha197 commented on resources/i18n/en.json:
```diff
@@ -579,6 +579,7 @@
     "RoutesInfo": "Routes Info",
     "RoutingID": "Routing ID",
     "ScalingSettings": "Scaling Settings",
+    "SelectDeployment": "Select Deployment",
```
Fixed in ecdb7d171. All non-English language files now include the complete deployment namespace (120 flat keys + 93 launcher keys). English text is used as fallback where translations don't exist yet, while previously translated keys (e.g., SelectDeployment) are preserved.

This approach:
- Ensures all deployment strings are present in every language file (addressing the review concern)
- Uses English as fallback until proper translations can be added incrementally
- Follows common i18n practices for large translation sets
- Allows i18next to properly resolve keys in all languages

agatha197 commented:
I'll handle this with new spec file.
